config/v1/metric_descriptor.proto

```diff
@@ -0,0 +1,171 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package config.v1;
+
+import "label_descriptor.proto";
+
+// Defines a metric type and its schema.
+//
+// A given metric is described by a set of labels. These labels represent the
+// different dimensions to associate with the metric. A given metric holds a
+// unique value for potentially any combination of these labels.
+//
+// Metric values can be integers, floating point numbers, strings, or booleans.
+//
+// Here's an example metric descriptor defined in a YAML configuration
+// file:
+//
+//    metrics:
```

Comment:
I am thinking of a metric definition from the standpoint of entire istio, not just mixer.

```
- name: request_size
       elementary_measurement: request.size 
       metric_kind: DELTA
       value_type: INT64
       labels:
      - key: consumer_id
      - key: status_code
      - key: service_id
- name: bytes_transferred
       elementary_measurement: request.size + response.size
       metric_kind: DELTA
       value_type: INT64
       labels:
      - key: consumer_id
      - key: status_code
      - key: service_id
```
Disregarding proxy side aggregation, here is how this information is used jointly by mixer and the proxy.

proxy needs to know
1. The union of all labels to send across all metrics  [consumer_id, status_code, service_id]
2. The union of elementary_measurements needed by the metrics  [request.size, response.size]

Report call will contain all the above information. At this point mixer is able to assemble the actual metrics and add labels. 

.... However the same declarative metric definition can be used to fully compute the metric on proxy side.